packages/bpk-component-image/src/BpkBackgroundImage.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import { BpkSpinner } from 'bpk-component-spinner';
+
+import STYLES from './bpk-background-image.scss';
+
+const getClassName = cssModules(STYLES);
+
+class BpkBackgroundImage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.onBackgroundImageLoad = this.onBackgroundImageLoad.bind(this);
+    this.startImageLoad = this.startImageLoad.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.props.inView) {
+      this.startImageLoad();
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (!this.props.inView && newProps.inView) {
+      this.startImageLoad();
+    }
+  }
+
+  onBackgroundImageLoad() {
+    if (this.props.onLoaded) {
+      this.props.onLoaded();
+    }
+  }
+
+  startImageLoad() {
+    const trackImage = new Image();
+    trackImage.src = this.props.src;
+    trackImage.onload = this.onBackgroundImageLoad;
+  }
+
+  render() {
+    const { backgroundColour, children, className, inView, loading, onLoaded, src, style, title, ...rest } = this.props;
+
+    const classNames = [getClassName('bpk-background-image')];
+    const spinnerClassNames = [getClassName('bpk-background-image__spinner')];
+    const contentClassNames = [getClassName('bpk-background-image__content')];
+
+    if (!loading) {
+      spinnerClassNames.push(getClassName('bpk-background-image__spinner--hide'));
+      contentClassNames.push(getClassName('bpk-background-image__content--show'));
+    }
+
+    if (className) {
+      classNames.push(className);
+      contentClassNames.push(className);
+    }
+
+    const displayedElements = [
+      <div key="spinner" className={spinnerClassNames.join(' ')}>
+        <BpkSpinner />
+      </div>,
+    ];
+
+    if (inView) {
+      displayedElements.push(
+        <div
+          key="background-div"
+          className={contentClassNames.join(' ')}
+          style={{
+            backgroundImage: loading ? '' : `url(${src})`,
+            ...style,
+          }}
+          {...rest}
+        >
+          {!loading ? children : (null)}
+        </div>,
+      );
+    }
+
+    return (
+      <div
+        style={{ background: backgroundColour, ...style }}
```

Comment:
This is a bit weird. If a consumer supplied a `style` prop, this is added both to the container AND the element containing the background image?